src/pages/cats.js

```diff
@@ -0,0 +1,760 @@
+  adblockerWhitelistingURL,
+  externalHelpAllAppsURL,
+  facebookPageURL,
+  twitterPageURL,
+  tiktokPageURL,
+  instagramPageURL,
+  homeURL,
+} from 'src/utils/navigation'
+import logoWhite from 'src/img/logo-white.svg'
+import cat1 from 'src/img/cats/1.png'
+import cat2 from 'src/img/cats/2.png'
+// Reviewer images
+import reviewImgAbbyH from 'src/img/reviews/abby_h.png'
+import reviewImgAlfonzoG from 'src/img/reviews/alfonzo_g.png'
+import reviewImgAnaL from 'src/img/reviews/ana_l.png'
+import reviewImgBenjaminK from 'src/img/reviews/benjamin_k.png'
+import reviewImgCameronB from 'src/img/reviews/cameron_b.png'
+import reviewImgChaseR from 'src/img/reviews/chase_r.png'
+import reviewImgJohnathanB from 'src/img/reviews/johnathan_b.png'
+import reviewImgLucasN from 'src/img/reviews/lucas_n.png'
+import reviewImgShayneW from 'src/img/reviews/shayne_w.png'
+import reviewImgTobyS from 'src/img/reviews/toby_s.png'
+import localStorageMgr from 'src/utils/local-storage'
+import { lightestTextColor } from 'src/themes/theme'
+import {
+  STORAGE_REFERRAL_DATA_REFERRING_CHANNEL,
+  STORAGE_REFERRAL_DATA_REFERRING_USER,
+  STORAGE_NEW_USER_IS_TAB_V4_BETA,
+} from 'src/utils/constants'
+import { getUrlParameterValue } from 'src/utils/location'
+import Divider from '@material-ui/core/Divider'
+import redirect from 'src/utils/redirect'
+const canonicalURL = getAbsoluteURL(homeURL)
+const useStyles = makeStyles(() => ({
+  whiteFont: {
+    color: '#fff',
+  },
+  sectionSplit: {
+    display: 'flex',
+    minHeight: '460px',
+    alignItems: 'center',
+  },
+  halfPage: {
+    width: '50%',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  logo: {
+    width: 40,
+    height: 40,
+  },
+  logoContainer: { flex: 1 },
+  title: {
+    fontSize: '75px',
+    color: 'black',
+    fontWeight: 'bold',
+    textAlign: 'center',
+    marginBottom: '30px',
+  },
+  address: {
+    margin: 0,
+    fontStyle: 'italic',
+    fontSize: 12,
+  },
+}))
+const Cats = ({ pageContext }) => {
+  const cx = useStyles()
+  // store referrer id
+  useEffect(() => {
+    let referrerId = null
+    // Check for a referrer's vanity URL.
+    if (pageContext && pageContext.referrer) {
+      referrerId = pageContext.referrer.id
+    } else {
+      const paramRefId = parseInt(getUrlParameterValue('r'))
+      if (!isNaN(paramRefId)) {
+        referrerId = paramRefId
+      }
+    }
+    referrerId &&
+      localStorageMgr.setItem(
+        STORAGE_REFERRAL_DATA_REFERRING_CHANNEL,
+        referrerId
+      )
+  }, [])
+  useEffect(() => {
+    const userReferrerId = getUrlParameterValue('u')
+    if (userReferrerId !== null && userReferrerId !== undefined) {
+      localStorageMgr.setItem(
+        STORAGE_REFERRAL_DATA_REFERRING_USER,
+        userReferrerId
+      )
+    }
+  }, [])
+  const installButton = (
+    <InstallButton
+      onBeforeInstall={() => {
+        localStorageMgr.setItem(STORAGE_NEW_USER_IS_TAB_V4_BETA, 'true')
```

Comment:
Please add a test for this. You can manually invoke the prop and confirm it behaves as expected.